include/dba.php

```diff
@@ -355,73 +429,98 @@ public function dbg($dbg) {
 
 	public function escape($str) {
 		if ($this->db && $this->connected) {
-			if ($this->mysqli) {
-				return @$this->db->real_escape_string($str);
-			} else {
-				return @mysql_real_escape_string($str,$this->db);
+			switch ($this->driver) {
+				case 'pdo':
+					return substr(@$this->db->quote($str, PDO::PARAM_STR), 1, -1);
```

Comment:
I would not like it surpressing errors/warnings with `@` which makes it much harder with debugging (you really then have to look blindly why this or that is not working). Better is, to remove (or even ban on PHP's side) and on productive servers use `error_reporting(0);` and on development servers `error_reporting(E_ALL | E_STRICT);` (the later for a bit older PHP versions).